File: Wrp/php/src/com/amivoice/wrp/WrpListener.php
<?php

namespace com\amivoice\wrp;

interface WrpListener {
	function utteranceStarted($startTime);
	function utteranceEnded($endTime);
	function resultCreated();
	function resultUpdated($result);
	function resultFinalized($result);
	function eventNotified($eventId, $eventMessage);
	function TRACE($message);
}

?>

File: Wrp/python/src/com/amivoice/wrp/WrpListener.py
class WrpListener:
	#@abstractmethod
	def utteranceStarted(startTime):
		pass
	#@abstractmethod
	def utteranceEnded(endTime):
		pass
	#@abstractmethod
	def resultCreated():
		pass
	#@abstractmethod
	def resultUpdated(result):
		pass
	#@abstractmethod
	def resultFinalized(result):
		pass
	#@abstractmethod
	def eventNotified(eventId, eventMessage):
		pass
	#@abstractmethod
	def TRACE(message):
		pass
